src/multiplex/client.rs

```diff
@@ -138,21 +243,138 @@ where
     }
 }
 
-// ===== ClientInner =====
+// ===== Pending =====
 
-struct Pending<Tag, Item> {
+/// A type used to track in-flight requests.
+///
+/// Each pending response contains an associated `Tag` that is provided
+/// by the [`TagStore`], which is used to uniquely identify a request/response pair.
+pub struct Pending<Tag, Response> {
     tag: Tag,
```

Comment:
Do we need to keep the `tag` stored in the `Pending` structure or should we remove it? It seems that the only reason it is here is to be able to compare it in the implementation using `VecDeque`. If I implement a `PendingStore` using a `HashMap` for example, i would need to clone it so that it could be used as a HashMap key, and it would be both in the key and the value.

Maybe we should remove it, and pass it to sent explicitly, like
`fn sent(self: Pin<&mut Self>, pending: Pending<T::Tag, T::Ok>, transport: Pin<&mut T>);`. Then the PendingStore could make its own decision about how it wants to use it to match tags of inflight requests with the incoming tag.

Reply:
Yeah that seems like a fair change let me look into that today.